frontend/src/services/FeatureFlagManager.ts

```diff
@@ -238,14 +264,14 @@ export class FeatureFlagsManager {
   }
 
   /**
-   * Toggle feature flag
+   * Toggle feature flag on cookie
    */
-  setFeatureFlag(featureName: string, enabled: boolean): void {
+  setFeatureFlagCookie(featureName: string, enabled: boolean): void {
     if (!this.isValidFeatureFlag(featureName)) {
       throw new Error(`\`${featureName}\` is not a valid feature flag`);
     }
     const featureFlags = {
-      ...this.featureFlags,
+      ...this.featureFlagsCookie,
```

Comment:
this is a change to the existing functionality - previously whenever we set the feature flags cookie it would enshrine the default feature flag values in the cookie, but that functionality conflicts with idea of honoring flags set in the environment and using cookies as an override. Setting a flag on the cookie should, in this case, only set the single flag value specified